mLRS/Common/hal/timer.h

```diff
@@ -17,14 +17,27 @@
 
 #define SYSTICK_DELAY_MS(x)       (uint16_t)(((uint32_t)(x)*(uint32_t)1000)/SYSTICK_TIMESTEP)
 
+// Two variables are required to avoid race with HAL_IncTick() when tasks may take longer than one tick
+uint32_t doSysTask_done = 0; // Never changed in ISR; incremented when (uwTick != doSysTask_done)
 
-volatile uint32_t doSysTask = 0;
+void resetSysTask()
+{
+    doSysTask_done = uwTick;
+}
 
+volatile bool doSysTask()
+{
+    if (uwTick != doSysTask_done)
+        {
```

Comment:
here?
it seems I'm saying something very difficult :)

Reply:
Ahh.  OK.  I thought you meant below in the same file.